multimodal/src/autogluon/multimodal/utils.py

```diff
@@ -2142,3 +2150,442 @@ def compute_num_gpus(config_num_gpus: Union[int, float, List], strategy: str):
         num_gpus = 1
 
     return num_gpus
+
+
+_URL_REGEX = re.compile(
+    r"^(?:http|ftp)s?://"  # http:// or https://
+    r"(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|"  # domain...
+    r"localhost|"  # localhost...
+    r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"  # ...or ip
+    r"(?::\d+)?"  # optional port
+    r"(?:/?|[/?]\S+)$",
+    re.IGNORECASE,
+)
+
+
+def is_url(url_like: str):
+    """
+    Check if a path is url or local.
+
+    Parameters
+    ----------
+    url_like
+        A provided path.
+
+    Returns
+    -------
+    A boolean indicate if the path is an url.
+    """
+    if not isinstance(url_like, str):
+        return False
+    return re.match(_URL_REGEX, url_like) is not None
+
+
+def from_voc(
+    root: str,
+    splits: Optional[Union[str, tuple]] = None,
+    exts: Optional[Union[str, tuple]] = (".jpg", ".jpeg", ".png"),
+):
+    """
+    Construct dataframe from pascal VOC format. Modified from gluon cv.
+    Normally you will see a structure like:
+    ├── VOC2007
+    │   ├── Annotations
+    │   ├── ImageSets
+    |   |   ├── Main
+    |   |   |   ├── train.txt
+    |   |   |   ├── test.txt
+    │   ├── JPEGImages
+
+    Parameters
+    ----------
+    root
+        The root directory for VOC, e.g., the `VOC2007`. If an url is provided, it will be downloaded and extracted.
+    splits
+        If given, will search for this name in `ImageSets/Main/`, e.g., ('train', 'test')
+    exts
+        The supported image formats.
+
+    Returns
+    -------
+    A dataframe with columns "image", "rois", and "image_attr".
+    """
+    if is_url(root):
+        root = download(root)
+    rpath = Path(root).expanduser()
+    img_list = []
+    class_names = set()
+    if splits:
+        logger.debug("Use splits: %s for root: %s", str(splits), root)
+        if isinstance(splits, str):
+            splits = [splits]
+        for split in splits:
+            split_file = rpath / "ImageSets" / "Main" / split
+            if not split_file.resolve().exists():
+                split_file = rpath / "ImageSets" / "Main" / (split + ".txt")
+            if not split_file.resolve().exists():
+                raise FileNotFoundError(split_file)
+            with split_file.open(mode="r") as fi:
+                img_list += [line.split()[0].strip() for line in fi.readlines()]
+    else:
+        logger.debug(
+            "No split provided, use full image list in %s, with extension %s", str(rpath / "JPEGImages"), str(exts)
+        )
+        if not isinstance(exts, (list, tuple)):
+            exts = [exts]
+        for ext in exts:
+            img_list.extend([rp.stem for rp in rpath.glob("JPEGImages/*" + ext)])
+    print(img_list)
```

Comment:
Will there be too many outputs for this print? Maybe we can just print its length

Reply:
Done.